web/js/userTasksReport.js

```diff
@@ -285,12 +348,22 @@ Ext.onReady(function () {
             handler: findTasks,
         }],
     });
+    /* Allow listing of user tasks of other users only for an admin user */
+    if ( admin == "" ) {
+        Ext.getCmp('userLogin').setDisabled(true);
+    }
 
     /* Handler to invoke the search service */
     function findTasks () {
-                var baseParams = {
-                    'userId': userId,
-                };
+                var baseParams = {};
+
+                if (Ext.getCmp('userLogin').baseParams.userId == ""){
+                    // If no value was set in the combobox, default to the logged in user
+                    baseParams.userId = userId;
+                } else {
```

Comment:
Take a look at the time fields in the tasks screen, they are compulsory and tasks cannot be saved without them. It's a similar case.